### PR TITLE
refactor(Subscription): remove the type parameter from `Subscription`

### PR DESCRIPTION
**Pull request by saneyuki**
This type parameter is not used actually.

### CODE REVIEW

File: src/Subscription.ts [hunk 1]
```diff
@@ -1,14 +1,14 @@
 import {noop} from './util/noop';
 
-export class Subscription<T> {
```
Comment (staltz) [inline src/Subscription.ts]: Hmm, makes sense to me. :+1:

**Comment by kwonoj** [comment]
Yes, this seems quite legit to me too.

File: src/operator/groupBy-support.ts [hunk 2]
```diff
@@ -3,8 +3,8 @@ import {Subject} from '../Subject';
 import {Subscriber} from '../Subscriber';
 import {Observable} from '../Observable';
 
-export class RefCountSubscription<T> extends Subscription<T> {
-  primary: Subscription<T>;
+export class RefCountSubscription<T> extends Subscription {
```
Comment (luisgabriel) [inline src/operator/groupBy-support.ts]: I think you can also remove the type parameter from `RefCountSubscription<T>`. What do you think?

File: src/operator/groupBy-support.ts [hunk 3]
```diff
@@ -44,7 +44,7 @@ export class GroupedObservable<T> extends Observable<T> {
   }
 }
 
-export class InnerRefCountSubscription<T> extends Subscription<T> {
+export class InnerRefCountSubscription<T> extends Subscription {
```
Comment (luisgabriel) [inline src/operator/groupBy-support.ts]: Same here.

**Comment by saneyuki** [comment]
> I think you can also remove the type parameter from RefCountSubscription<T>. What do you think?
> Same here.

okay. I removed them!

**Comment by luisgabriel** [comment]
LGTM.

**Comment by kwonoj** [comment]
Change looks good to me too.

**Comment by saneyuki** [comment]
rebased! I resolved the conflict

**Comment by blesh** [comment]
LGTM. cc/ @kwonoj

**Comment by kwonoj** [comment]
I'll check in this today.

**Comment by kwonoj** [comment]
Actually, doing it now...

**Comment by kwonoj** [comment]
Merged with a8941d7, thanks @saneyuki . Appreciate for effort along with nice catch.

**Pull request closed without merging**